product-accelerators/apim/components/org.wso2.healthcare.apim.backendauth/src/main/java/org/wso2/healthcare/apim/backendauth/PrivateKeyJWTBackendAuthenticator.java

```diff
@@ -0,0 +1,313 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.healthcare.apim.backendauth;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.nimbusds.jose.*;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpEntity;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.wso2.carbon.core.util.KeyStoreManager;
+import org.wso2.carbon.utils.CarbonUtils;
+import org.wso2.healthcare.apim.backendauth.tokenmgt.Token;
+import org.wso2.healthcare.apim.backendauth.tokenmgt.TokenManager;
+import org.wso2.healthcare.apim.core.OpenHealthcareException;
+import org.wso2.healthcare.apim.core.OpenHealthcareRuntimeException;
+import org.wso2.healthcare.apim.core.config.BackendAuthConfig;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.security.interfaces.RSAPrivateKey;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Mediator to authenticate with the backend using private key JWT.
+ */
+public class PrivateKeyJWTBackendAuthenticator extends BackendAuthenticator {
+
+    private static final Log log = LogFactory.getLog(PrivateKeyJWTBackendAuthenticator.class);
+    private static final JsonParser parser = new JsonParser();
+
+    public PrivateKeyJWTBackendAuthenticator() throws OpenHealthcareException {
+    }
+
+    public boolean mediate(MessageContext messageContext) {
+
+        log.info("PrivateKeyJWTBackendAuthenticator mediator is started.");
+
+        String accessToken;
+        String tokenEndpoint = backendAuthConfig.getAuthEndpoint();
+        String keyAlias = backendAuthConfig.getPrivateKeyAlias();
+        char[] keyStorePass = CarbonUtils.getServerConfiguration().getFirstProperty(
+                "Security.KeyStore.Password").toCharArray();
```

Comment:
Use constants

Reply:
fixed in [b8da50d](https://github.com/wso2/healthcare-accelerator/pull/7/commits/b8da50dbe832d38ee8259489c5c03798ba7ca81d)